zephyr/include/sof/lib/regions_mm.h

```diff
@@ -10,31 +10,240 @@
 
 #include <adsp_memory.h>
 #include <adsp_memory_regions.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
+#include <sof/list.h>
 #include <zephyr/drivers/mm/system_mm.h>
-#include <zephyr/sys/mem_blocks.h>
 #include <zephyr/init.h>
-#include <ipc/topology.h>
+#include <zephyr/sys/mem_blocks.h>
 
-/*
- * Struct containing information on virtual memory heap.
- * Information about allocated physical blocks is stored in
- * physical_blocks_allocators variable and uses zephyr memblocks api.
+/* Dependency on ipc/topology.h created due to memory capability definitions
+ * that are defined there
+ */
+
+/* Defines maximum amount of memory block allocators in one heap
+ * since minimum realistic block should be DCASHE_LINE size
+ * and block sizes in allocator must be powers of 2
+ * so logicly it gives limited block sizes
+ * eg allocator of block size 64 128 256 512 2048 4096 1024 8192
+ * would end up in 8 allocators.
+ * It is expected that allocations bigger than that would
+ * either be spaned on specifically configured heap
+ */
+#define MAX_MEMORY_ALLOCATORS_COUNT 8
+
+/* list of virtual_memory_heap objects created */
+struct list_item vm_heaps;
+/* list of vmh_active_physical_mapping objects created */
+struct list_item physical_mappings;
+
+/** @struct virtual_memory_heap
+ *
+ *  @brief This structure holds all information about virtual memory heap
+ *  it aggregates information about its allocations and
+ *  physical mappings.
+ *
+ *  @var node generic list member used for list operations
+ *  @var virtual_region pointer to virtual region information, it holds its
+ *  attributes size and beginning ptr provided by zephyr.
+ *  @var physical_blocks_allocators[] a table of block allocators
+ *  each representing a virtual regions part in blocks of a given size
+ *  governed by sys_mem_blocks API.
+ *  @var active_allocations list of current heap allocations across
+ *  all its allocators (sys_mem_blocks) saved in vmh_active_physical_mapping
+ *  struct.
+ *  TO DO: mem_blocks API should be broadened to allow saving
+ *  allocations to a bitarray.
+ *  @var span configuration value deciding if heap allocations
+ *  will be contiguous or single block.
  */
 struct virtual_memory_heap {
-	/* zephyr provided virtual region */
+	struct list_item node;
 	struct sys_mm_drv_region *virtual_region;
-	/* physical pages allocators represented in memblocks */
-	struct sys_multi_mem_blocks physical_blocks_allocators;
-	/* SOF memory capability */
-	uint32_t memory_caps;
+	struct sys_mem_blocks
+	    *physical_blocks_allocators[MAX_MEMORY_ALLOCATORS_COUNT];
```

Comment:
in your previous version this was 1 line and it was 79 characters long. Please keep 1 line.

Reply:
It is over 80 with change to define name.

Reply:
SOF also uses 100 as a line length limit now

Reply:
Switched config in my clang formater - redone whole commit in accordance to 100 lines